Core/Src/monitor.c

```diff
@@ -139,6 +147,71 @@ void vPedalsMonitor(void* pv_params)
 		// MAX_ADC_VAL_12B) && 	!is_timer_active(&diff_timer)) 	start_timer(&diff_timer,
 		// PEDAL_FAULT_TIME); else 	cancel_timer(&diff_timer);
 
+
+		/* Evaluate accelerator faults */
+		if (is_timer_expired(&oc_timer))
+			//todo queue fault
+			continue;
+		else if ((adc_data[ACCELPIN_1] == MAX_ADC_VAL_12B || adc_data[ACCELPIN_2] == MAX_ADC_VAL_12B) &&
+			!is_timer_active(&oc_timer))
+			start_timer(&oc_timer, PEDAL_FAULT_TIME);
+		else
+			cancel_timer(&oc_timer);
+		
+		if (is_timer_expired(&sc_timer))
+			//todo queue fault
+			continue;
+		else if ((adc_data[ACCELPIN_1] == 0 || adc_data[ACCELPIN_2] == 0) &&
+			!is_timer_active(&sc_timer))
+			start_timer(&sc_timer, PEDAL_FAULT_TIME);
+		else
+			cancel_timer(&sc_timer);
+
+		if (is_timer_expired(&diff_timer))
+			//todo queue fault
+			continue;
+		else if ((adc_data[ACCELPIN_1] - adc_data[ACCELPIN_2] > PEDAL_DIFF_THRESH * MAX_ADC_VAL_12B) &&
+			!is_timer_active(&diff_timer))
+			start_timer(&diff_timer, PEDAL_FAULT_TIME);
+		else
+			cancel_timer(&diff_timer);
+
+		/*Evaluate brake faults*/
+
+		/*Open circuit*/
+
+		if(is_timer_expired(&oc_timer))
```

Comment:
I think we need to review this faulting logic. Right now the timer will never actually reset itself if we fix the fault. See my implementation of the pedal reading for last year's car: https://github.com/Northeastern-Electric-Racing/Embedded_Code/blob/main/MPU/src/pedalHW.cpp

Reply:
reviewed the code and made it more verbose